app/models/region.rb

```diff
@@ -0,0 +1,36 @@
+class Region < ApplicationRecord
+  extend FriendlyId
+  friendly_id :name, use: :slugged
+  enum level: {
+    root: 0,
+    organization: 10,
+    state: 20,
+    facility_group: 30,
+    facility: 40
+  }
+
+  belongs_to :parent_region, polymorphic: true, optional: true
+  has_many :_child_regions, as: :parent_region, class_name: "Region"
+  has_many :_facility_groups, inverse_of: :parent_region, foreign_key: :parent_region_id
```

Comment:
Are these "private" associations because they're only here for compatibility until everything becomes a region?

Reply:
Yup you got it.